File: CPP/ByTier/C/1728_C_Digital_Logarithm.cpp
// Problem: CF 1728 C - Digital Logarithm
// https://codeforces.com/contest/1728/problem/C

/*
Purpose:
This code solves the problem of making two arrays similar by applying a "digital logarithm" operation,
which replaces each element with the length of its base-10 representation.
The approach uses a greedy strategy with a map to count and balance the frequency of digit lengths.

Algorithms/Techniques:
- Map-based frequency counting with digit length transformation
- Greedy balancing of frequencies to minimize operations
- Digit length calculation using string conversion

Time Complexity: O(n log M) where n is the number of elements and M is the maximum value (up to 10^9)
Space Complexity: O(n) for the map storage
*/

#include <stdlib.h>
#include <functional>
#include <iostream>
#include <map>
#include <string>
#include <utility>

using namespace std;
map<int, int, greater<int>> mp; // Map to store frequency of numbers and their digit lengths
int T, n, a, ans;

void solve() {
  cin >> n;
  ans = 0;
  mp.clear();
  
  // Read all n numbers from array 'a' and increment their count in map
  for (int i = 0; i < 2 * n; i++)
    cin >> a, mp[a] += i < n ? 1 : -1;
  
  // For each entry in the map, transform the key to its digit length and accumulate counts
  for (auto k : mp)
    mp[to_string(k.first).length()] += k.second, ans += abs(k.second);
  
  cout << ans << '\n';
}

int main() {
  cin >> T;
  while (T--)
    solve();
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByRound/1728/1728_C_Digital_Logarithm.py
# Problem: CF 1728 C - Digital Logarithm
# https://codeforces.com/contest/1728/problem/C

"""
Purpose: 
This code solves the problem of making two arrays similar by applying digital logarithm operations. 
The digital logarithm f(x) is defined as the number of digits in x (i.e., length of base-10 representation).
In each operation, we can either replace a[i] with f(a[i]) or f(b[i]) for some index i.
We aim to find the minimum number of such operations to make the two arrays similar (i.e., have the same multiset of elements after rearrangement).

Key Algorithm:
1. Count frequencies of each number in both arrays.
2. Compute difference in counts between arrays.
3. For numbers with more than one digit, we can potentially convert them to their digital logarithm (number of digits).
4. The operations required is the sum of absolute differences of counts, considering conversions where possible.

Time Complexity: O(n), where n is the size of arrays. Each element is processed once, and operations on Counter are efficient.
Space Complexity: O(n), due to storing counters and keys.
"""

from collections import Counter

for _ in range(int(input())):
    n = int(input())
    a = Counter(input().split())  # Count frequencies of elements in array a
    b = Counter(input().split())  # Count frequencies of elements in array b
    a.subtract(b)  # Subtract b's counts from a's; positive values mean excess in a, negative in b
    keys = list(a.keys())
    ans = 0
    num1 = 0
    for k in keys:
        if len(k) > 1:  # If current number has more than one digit
            a[str(len(k))] += a[k]  # Transfer count to its digital logarithm
            ans += abs(a[k])  # Add operations needed for this conversion
            a[k] = 0  # Clear the original key
    for k in a:
        if k != "1":  # For all keys except 1 (since f(1) = 1)
            ans += abs(a[k])  # Add the remaining differences
    print(ans)


# https://github.com/VaHiX/CodeForces/